octoprint_v8theme/static/js/v8theme.js

```diff
@@ -14,20 +164,269 @@ $(function() {
+                position: {
+                    my: "left top+10",
+                    at: "center-15 bottom",
+                    collision: "flipfit"
+                }
+            });
         };
 
         self.onAfterTabChange = function(current, previous) {
             if (current != "#control") {
                 return;
             }
-            self.temp.updatePlot();
+            self.temperature.updatePlot();
         }
+        
+        self.onStartupComplete = function() {
+            self.parseCustomControls();
+
+            $("#control").appendTo("#control_main .accordion-inner");
+            $("#terminal_main small.pull-left span").css({
+                "display": "block",
+                "text-align": "right"
+            }).appendTo("#terminal_main small.pull-right");
+            $(".sd-trigger a:first").html('<i class="icon-file"></i>');
+            $("#temp").remove();
+            $("#term").remove();
+            $("#webcam_container, #control_main div[data-bind*='keycontrolPossible']").remove();
+            // Manage extra contents of .tab-content 
+            $("#gcode").remove();
+            var tabContentHTML = $(".main-content-wrapper").html().replace(/<!-- ko allowBindings: false -->|<!-- \/ko -->|<!-- ko allowBindings: true -->/g, "");
+            if (tabContentHTML.trim().length) {
+              $(".main-content-wrapper").after("<div id='Additional_wrapper' class='accordion-group'> <div class='accordion-heading'> <a class='accordion-toggle' data-toggle='collapse' data-target='#Additional_main'><i class='icon-info-sign'></i> Additional Controls </a> </div> <div id='Additional_main' class='accordion-body collapse in'> <div class='accordion-inner'> </div></div></div></div>");
+              $(".main-content-wrapper").appendTo("#Additional_main .accordion-inner");
+            } else {
+              $(".main-content-wrapper").remove();
+            }
+        };
 
+        var lastScrollTop = 0;
+        $("#terminal-output").scroll(function(event) {
+            var st = $(this).scrollTop();
+            if (st < lastScrollTop && self.terminal.autoscrollEnabled()) {
+                self.terminal.autoscrollEnabled(false);
+            } else if (st > lastScrollTop) {
+                if ($(this).scrollTop() + $(this).innerHeight() >= this.scrollHeight) {
+                    self.terminal.autoscrollEnabled(true);
+                }
+            }
+            lastScrollTop = st;
+        });
     }
 
     OCTOPRINT_VIEWMODELS.push([
         V8ThemeViewModel,
-        ["temperatureViewModel"],
-        ""
+        ["temperatureViewModel", "terminalViewModel"],
+        []
     ]);
 });
+function checkWidth() {
+    if ($(window).width() > 767) {
+        $(".octoprint-container .span4").first().removeClass("mt0");
+    } else if ($(window).width() <= 767 && $(".nav-collapse").hasClass("in")) {
+        $(".octoprint-container .span4").first().addClass("mt0");
+    }
+}
```

Comment:
I think this could be accomplished with CSS.  You usually want to avoid JavaScript handlers of scroll and resize events if you can since they can't be run in the background and significantly slow down the UI.